streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -266,13 +272,13 @@ public void suspend() {
             case CREATED:
                 log.info("Suspended created");
                 transitionTo(State.SUSPENDED);
-
+                setTimeCurrentIdlingStarted();
```

Comment:
Actually I mean getting a helper like:
```
void transitToSuspend() {
  log.info("Suspended {}", state()); 
  transitionTo(State.SUSPENDED);
  timeCurrentIdlingStarted = Optional.of(System.currentTimeMillis());
}
```
since the operation is the same for all 3 transition calls

Reply:
sounds good